platform/security/platform-security-core-api/src/main/java/ddf/security/common/audit/SecurityLogger.java

```diff
@@ -76,30 +80,52 @@ private static void requestIpAndPortAndUserMessage(
   private static void requestIpAndPortAndUserMessage(
       Subject subject, Message message, StringBuilder messageBuilder) {
     String user = getUser(subject);
-    if (message != null) {
+    messageBuilder.append(SUBJECT).append(user);
+    appendConditionalAttributes(subject, messageBuilder);
+
+    if (message == null) {
+      messageBuilder.append(" ");
+    } else {
       HttpServletRequest servletRequest =
           (HttpServletRequest) message.get(AbstractHTTPDestination.HTTP_REQUEST);
       // pull out the ip and port of the incoming connection so we know
       // who is trying to get access
       if (servletRequest != null) {
         messageBuilder
-            .append(SUBJECT)
-            .append(user)
             .append(" Request IP: ")
             .append(servletRequest.getRemoteAddr())
             .append(", Port: ")
             .append(servletRequest.getRemotePort())
             .append(" ");
       } else if (MessageUtils.isOutbound(message)) {
         messageBuilder
-            .append(SUBJECT)
-            .append(user)
             .append(" Outbound endpoint: ")
             .append(message.get(Message.ENDPOINT_ADDRESS))
             .append(" ");
       }
-    } else {
-      messageBuilder.append(SUBJECT).append(user).append(" ");
+    }
+  }
+
+  /**
+   * Appends any additional attributes as defined in the comma-delimited system property {@link
+   * #EXTRA_ATTRIBUTES_PROP}.
+   *
+   * @param subject the subject of the logging request
+   * @param messageBuilder buffer to which to append attribute text, if any
+   */
+  private static void appendConditionalAttributes(Subject subject, StringBuilder messageBuilder) {
+    String attributes = System.getProperty(EXTRA_ATTRIBUTES_PROP);
+    if (subject == null) {
+      subject = ThreadContext.getSubject();
+    }
+    if (attributes != null) {
```

Comment:
✏️ Moderately cheaper and simpler to read if you remove this test here, add an `if (attributes == null)` test on line 118, and then just `return` then. You'll save the check on `subject` and potential local store.